solidity/contracts/proxy/DepositFactory.sol

```diff
@@ -51,9 +53,9 @@ contract DepositFactory is CloneFactory, TBTCSystemAuthority{
     ) public onlyTbtcSystem {
         masterDepositAddress = _masterDepositAddress;
         tbtcDepositToken = TBTCDepositToken(_tbtcDepositToken);
-        tbtcSystem = _tbtcSystem;
-        tbtcToken = _tbtcToken;
-        feeRebateToken = _feeRebateToken;
+        tbtcSystem = TBTCSystem(_tbtcSystem);
```

Comment:
Sort of in the same vein, can we bubble this typing up to the parameters for `setExternalDependencies` as well?